types/Cargo.toml

```diff
@@ -18,3 +18,7 @@ log = { version = "0.4", default-features = false }
 serde = { version = "1", default-features = false, features = ["derive"] }
 serde_json = { version = "1", default-features = false, features = ["alloc", "raw_value", "std"] }
 thiserror = "1.0"
+# TODO: a bit sad to have to pull in soketto to handle errors
+soketto = "0.4"
```

Comment:
yeah, I don't know how handle those without removing the shared error type.

but maybe a good idea to kill the big error type and that each crates has to define their own error type?

Reply:
Won't that make it tricky to share code between the servers?

Reply:
Yeah, unless you come up with super nice generic error type that both can use but then you need some extra boiler-plate.

Like `map.err(|e| Error::Transport(Box::new(e))`

Reply:
I can't think of a better way other than adding features, which I'm not sure is such a good idea.

Reply:
ok, you mean making `soketto` and `hyper` optional dependencies?

That's ok but why not let each crate implement implement the `From<TransportError> for Error`?
Then we shouldn't bring in any extra dependencies but a little bit of "redundant code", like for client/server as long they use the same transport library

Reply:
> hat's ok but why not let each crate implement implement the `From<TransportError> for Error`?
> Then we shouldn't bring in any extra dependencies but a little bit of "redundant code", like for client/server as long they use the same transport library

https://github.com/paritytech/jsonrpsee/pull/334